juju/api.go

```diff
@@ -453,10 +362,11 @@ func PrepareEndpointsForCaching(info configstore.EnvironInfo, hostPorts [][]netw
 		preferIPv6 := maybePreferIPv6(info)
 		network.SortHostPorts(uniqueHPs, preferIPv6)
 
-		if addrConnectedTo.Value != "" {
-			return network.EnsureFirstHostPort(addrConnectedTo, uniqueHPs)
+		for _, addr := range addrConnectedTo {
+			if addr.Value != "" {
```

Comment:
i vaguely recall this from an earlier version of juju. not sure now why it was needed. i can remove and see if the issue re-appears